### PR TITLE
add preprocessor to convert notebook to code

### LINKED ISSUE #229: Preprocessor to generate python code from notebook
I think the current preprocessors for notebooks work by pickling and depickling the code.

Can we instead have a preprocessor that would work by generating a python file that strips out everything except

* imports
* class defining the train & predict methods

### PR DESCRIPTION
**Pull request by abcdefgs0324**
**What this PR does / why we need it**:
Add a preprocessor to generate python code from notebook
* filter out everything except the cells including `fairing:include-cell`
* add preprocessor to generate code from notebook and create an entrypoint

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #229

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/310)

### CODE REVIEW

**Comment by jinchihe** [comment]
/lgtm

@abcdefgs0324 But could you please confirm any updates needed for current examples?

**Comment by abcdefgs0324** [comment]
I think there is no needed update for current examples.

**Comment by karthikv2k** [comment]
@abcdefgs0324 Thanks for your contribution. 
can you add a unit test?

File: fairing/preprocessors/converted_notebook.py [hunk 1]
```diff
@@ -55,3 +74,61 @@ def preprocess(self):
...
+                    if not lines[-1]:
+                        # last line is already blank don't add another one
+                        continue
+            # strip in statements
+            if l.startswith("# In["):
+                continue
+            lines.append(l)
+
+        contents = "\n".join(lines)
+        converted_notebook = Path(self.notebook_file).with_suffix('.py')
```
Comment (karthikv2k) [inline fairing/preprocessors/converted_notebook.py]: can you do this with a tempfile or check for existing file with the same name? writing a .py in the same dir might overwrite something the user already has.

File: tests/unit/preprocessors/test_notebook.py [hunk 2]
```diff
@@ -3,8 +3,13 @@
 
 # # Test Notebook
```
Comment (karthikv2k) [inline tests/unit/preprocessors/test_notebook.py]: I mistakenly checked-in this file. Can you please delete this as this should be auto generated during the test?

Comment (abcdefgs0324) [reply tests/unit/preprocessors/test_notebook.py]: Sure, I will delete it.

File: fairing/preprocessors/converted_notebook.py (hunk 1 above)
Comment (abcdefgs0324) [reply fairing/preprocessors/converted_notebook.py]: Okay, I also updated the original preprocessor.

**Comment by jinchihe** [comment]
@karthikv2k Seems the CI/CD env has problem, pls see details in the ticket https://github.com/kubeflow/testing/issues/417

**Comment by jinchihe** [comment]
@abcdefgs0324 could you please rebase to resolve the testing problem? Thanks.

**Comment by abcdefgs0324** [comment]
@jinchihe Rebased.

**Pull request merged**